fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java

```diff
@@ -81,7 +81,21 @@ public static void analyze(FunctionCallExpr functionCallExpr) {
                 }
             }
         }
-
+        
+        if (fnName.getFunction().equals(FunctionSet.REGEXP)) {
+            if (functionCallExpr.getChildren().size() > 2) {
+                final String lowerParam = ((StringLiteral) functionCallExpr.getChild(2)).getValue().toLowerCase();
```

Comment:
Firstly, you need check the argument is a stringLiteral or it may throw a cast failed exception.